<?xml version="1.0" ?>
<launch>
  <arg name="namespace" default="/rasberry_coordination"/>

  <arg name="agent_file_id" default="false"/>
  <arg name="setup_file_id" default="false"/>

  <arg name="agent_file" value="$(find rasberry_coordination)/config/agent/$(arg agent_file_id).yaml"/>
  <arg name="setup_file" value="$(find rasberry_coordination)/config/setup/$(arg setup_file_id).yaml"/>


  <group ns="$(arg namespace)">
    <!-- run add_agent for a specific agent -->
    <node pkg="rasberry_coordination" name="add_agent" type="add_agent.py" args="$(arg agent_file) $(arg setup_file)"
          output="screen" if="$(eval arg('agent_file_id') != false)"/>

    <!-- run add_agent to monitor for any new agents from car -->
    <node pkg="rasberry_coordination" name="add_agent" type="add_agent.py" args=""
          output="screen" if="$(eval arg('agent_file_id') == false)"/>


  </group>

</launch>


